sdk/iot/provisioning/inc/az_iot_provisioning_client.h

```diff
@@ -247,15 +247,14 @@ AZ_NODISCARD az_result az_iot_provisioning_client_register_publish_topic_get(
  * @note The payload of the MQTT publish message should be empty.
  *
  * @param[in] client The #az_iot_provisioning_client to use for this call.
- * @param[in] request_id The request id. Must match a received
- *                       #az_iot_provisioning_client_register_response request_id.
+ * @param[in] register_response The received response.

```

Comment:
nit: I like having the type which is returned because it allows for linking to the struct. In the case:
```c
// The received #az_iot_provisioning_client_register_response* 
```
This way someone reading the docs has easy access to the docs for this parameter.